src/d/actor/d_a_ship.cpp

```diff
@@ -5,460 +5,4713 @@
+            /* SrcObjCo  SPrm    */ cCcD_CoSPrm_Set_e | cCcD_CoSPrm_IsOther_e| cCcD_CoSPrm_VsGrpAll_e,
+            /* SrcGObjAt Se      */ 0,
+            /* SrcGObjAt HitMark */ 0,
+            /* SrcGObjAt Spl     */ 0,
+            /* SrcGObjAt Mtrl    */ 0,
+            /* SrcGObjAt SPrm    */ 0,
+            /* SrcGObjTg Se      */ dCcG_SE_UNK23,
+            /* SrcGObjTg HitMark */ 0,
+            /* SrcGObjTg Spl     */ 0,
+            /* SrcGObjTg Mtrl    */ 0,
+            /* SrcGObjTg SPrm    */ 0,
+            /* SrcGObjCo SPrm    */ 0,
+        },
+        // cM3dGSphS
+        {
+            /* Center */ 0.0f, 0.0f, 0.0f,
+            /* Radius */ 75.0f,
+        },
+    };
+    
+    fopAcM_SetupActor(this, daShip_c);
+    
+    if (dComIfGs_isEventBit(0xF80) == 0) {
+        return cPhs_ERROR_e;
+    }
+    
+    res = dComIfG_resLoad(&mPhs, l_arcName);
+
+    if (((fopAcM_GetParam(this) >> 8) & 0xFF) == 1) {
+        m03E8 = 1.0f;
+        onStateFlg(daSFLG_UNK200_e);
+    }
+    else {
+        m03E8 = 0.001f;
+        offStateFlg(daSFLG_UNK200_e);
+    }
+
+    if (res == cPhs_COMPLEATE_e) {
+        if (fopAcM_entrySolidHeap(this, daShip_createHeap, 0x20000) == 0) {
+            return cPhs_ERROR_e;
+        }
+        
+        m0298->setFrame(m0298->getEndFrame());
+        
+        pModel = m0298->getModel();
+        pModelData = pModel->getModelData();
+        
+        pModel->setUserArea(reinterpret_cast<u32>(this));
+        
+        fopAcM_SetMtx(this, pModel->getBaseTRMtx());
+        
+        for (u16 jno = 0; jno < pModelData->getJointNum(); jno++) {
+            if ((jno == 10) || (jno == 5) || (jno == 7) || (jno == 6)) {
+                pModelData->getJointTree().getJointNodePointer(jno)->setCallBack(daShip_bodyJointCallBack);
+            }
+        }
+
+        for (u16 mno = 0; mno < pModelData->getMaterialNum(); mno++) {
+            pMaterial = pModelData->getMaterialNodePointer(mno);
+            for(u32 i = 0; i < 8; i++) {
+                J3DTexGenBlock* pTexGenBlock = pMaterial->getTexGenBlock();
+                J3DTexMtx* pTexMtx = pTexGenBlock->getTexMtx(i);
+                if (pTexMtx && pTexMtx->getTexMtxInfo().mInfo == 8) {
+                    m02A0 = pTexMtx;
+                    m02A4 = pTexMtx;
+                }
+            }
+        }
+
+        C_MTXOrtho(m02A8, 50.0f, -50.0f, -50.0f, 50.0f, 0.1f, 100000.0f);
+        
+        pModel = m029C->getModel();
+        pModelData = pModel->getModelData();
+        
+        pModel->setUserArea(reinterpret_cast<u32>(this));
+        
+        for (u16 jno = 0; jno < pModelData->getJointNum(); jno++) {
+            if ((jno == 8) || (jno == 10)) {
+                pModelData->getJointTree().getJointNodePointer(jno)->setCallBack(daShip_headJointCallBack0);
+            }
+            else if (!((jno != 2) && (jno != 3) && (jno != 4) && (jno != 5) && (jno != 6)) || (jno == 7)) {
```

Comment:
When there's a bunch of double negatives it probably means the whole condition can be inverted, I think it would be something like this
```suggestion
            else if (jno == 2 || jno == 3 || jno == 4 || jno == 5 || jno == 6 || jno == 7) {
```

Reply:
Yeah I tried inverting this condition but the asm generated is different. No idea why but I even noticed that when using certain inline functions in conditionals, using `&&` and `||` logical operators would change the asm that was generated. I'll leave a comment mentioning it.

Reply:
The type of the variable can also impact how the compiler optimizes these chained comparisons btw, but the u16 you have would normally already be correct for a joint no, so not sure what the issue could be here